ee/modules/030-cloud-provider-openstack/hooks/migrate/migrate_terraform_state.go

```diff
@@ -0,0 +1,190 @@
+		}
+
+		if !gjson.ValidBytes(terraformStateRaw) {
+			return fmt.Errorf("invalid JSON: %s", terraformStateRaw)
+		}
+
+		terraformState := gjson.ParseBytes(terraformStateRaw)
+		openstackV2Resources := terraformState.Get("resources.#(type==\"openstack_blockstorage_volume_v2\")#")
+
+		if len(openstackV2Resources.Array()) == 0 {
+			input.LogEntry.Infof("No old resources found. Migration is not needed for Secret/%s/%s.", TerraformStateNamespace, secret.ObjectMeta.Name)
+			continue
+		}
+
+		if !BackupSecret(backupSecretName, secret, TerraformStateNamespace, kubeCl, input) {
+			return fmt.Errorf("can't create backup for Secret/%s/%s. The migration process has been aborted", TerraformStateNamespace, secret.ObjectMeta.Name)
+		}
+
+		terraformResources := terraformState.Get("resources")
+		for i, terraformResource := range terraformResources.Array() {
+			if terraformResource.Get("type").String() == OpenstackV2ResourceType {
+				input.LogEntry.Infof("Found resourceType = %s with name %s. Index = %d. Modifying resource.", terraformResource.Get("type").String(), terraformResource.Get("name").String(), i)
+				terraformStateRaw, err = sjson.SetBytes(terraformStateRaw, fmt.Sprintf("resources.%d.instances.0.attributes.multiattach", i), nil)
+				if err != nil {
+					return err
+				}
+				terraformStateRaw, err = sjson.SetBytes(terraformStateRaw, fmt.Sprintf("resources.%d.instances.0.attributes.enable_online_resize", i), true)
+				if err != nil {
+					return err
+				}
+				terraformStateRaw, err = sjson.SetBytes(terraformStateRaw, fmt.Sprintf("resources.%d.type", i), OpenstackV3ResourceType)
+				if err != nil {
+					return err
+				}
+			}
+			for j, terraformDependency := range terraformResource.Get("instances.0.dependencies").Array() {
+				if strings.Contains(terraformDependency.String(), "openstack_blockstorage_volume_v2") {
+					input.LogEntry.Infof("Found dependency %s with old resource in resource = %s and type = %s. Modifying dependency.", terraformDependency.String(), terraformResource.Get("name").String(), terraformResource.Get("type").String())
+					terraformStateRaw, err = sjson.SetBytes(terraformStateRaw, fmt.Sprintf("resources.%d.instances.0.dependencies.%d", i, j), strings.ReplaceAll(terraformDependency.String(), OpenstackV2ResourceType, OpenstackV3ResourceType))
+					if err != nil {
+						return err
+					}
+				}
+			}
+		}
+
+		var newTerraformState bytes.Buffer
+		err = json.Indent(&newTerraformState, terraformStateRaw, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		secret.Data[TerraformStateDataKey] = newTerraformState.Bytes()
+		_, err = kubeCl.CoreV1().
+			Secrets(TerraformStateNamespace).
+			Update(context.TODO(), &secret, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+		input.LogEntry.Infof("Migration process completed for Secret/%s/%s.", TerraformStateNamespace, secret.ObjectMeta.Name)
+	}
+
+	input.LogEntry.Infof("Terraform migrate hook has been finished.")
```

Comment:
i think we should remove this log entry

Reply:
I've removed this line: `input.LogEntry.Infof("Terraform migrate hook has been finished.")`